crunz

```diff
@@ -59,5 +59,6 @@ $loadAutoloader();
 $application = new Application(
     'Crunz Command Line Interface',
     InstalledVersions::getPrettyVersion('crunzphp/crunz') ?? '1.0.x-dev',
+    ($basePath = getenv('CRUNZ_BASE_CACHE_PATH')) !== false ? $basePath : null
```

Comment:
That logic (and `$baseCachePath ?? \sys_get_temp_dir();` from `Application`) should be in separate class, for example:
```php
final class CacheDirectoryFactory
{
    public function generate(): Path;
}
```
And then, this class can be used in `Application` directly, same as `EnvFlags`.
WDYT?

Reply:
> Do you mean something like following?

Yep

> I prefer being able to change the state of an object by being able to inject what I need. So I would prefer to give this configuration from the outside:

Actually me too :) But in this case, when `Application` instance doesn't come from DI container, it will be easier do it without constructor injection. Of course, it could be changed anytime, but we will see how it goes

Reply:
Okay, I will do the suggested adjustment. :)

Reply:
Is the current implementation like you imagined?